src/main/java/apoc/util/Util.java

```diff
@@ -210,40 +215,71 @@ public static Double toDouble(Object value) {
     }
 
     public static long toLong(Object value) {
-        if (value instanceof Number) return ((Number)value).longValue();
+        if (value instanceof Number) return ((Number) value).longValue();
         return Long.parseLong(value.toString());
     }
 
     public static URLConnection openUrlConnection(String url, Map<String, Object> headers) throws IOException {
+
         URL src = new URL(url);
-        URLConnection con = src.openConnection();
+        HttpURLConnection con = (HttpURLConnection) src.openConnection();
```

Comment:
we cannot assume that it is an HTTPURLConnection
we have to check after the fact for redirect handling